lib/ansible/modules/cloud/vmware/vmware_guest.py

```diff
@@ -510,6 +518,55 @@ def is_equal_cdrom(vm_obj, cdrom_device, cdrom_type, iso_path):
                     cdrom_device.connectable.startConnected and
                     (vm_obj.runtime.powerState != vim.VirtualMachinePowerState.poweredOn or cdrom_device.connectable.connected))
 
+    @staticmethod
+    def create_sio_controller():
+        sio_ctl = vim.vm.device.VirtualDeviceSpec()
+        sio_ctl.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
+        sio_ctl.device = vim.vm.device.VirtualSIOController()
+        sio_ctl.device.deviceInfo = vim.Description()
+        sio_ctl.device.busNumber = 0
```

Comment:
Any time you add a controller you need to use a global controller key index to ensure uniqueness. Otherwise if multiple controllers are being added in a single edit operation, and you add a device to one of those controllers, vmware won't know which controller the device belongs to.
There's a pending PR to fix this issue for CD-ROMs (https://github.com/ansible/ansible/pull/37869, specifically commit https://github.com/ansible/ansible/pull/37869/commits/d21a53a4c2d830c970060563b9862e36ec667d1a), but the same applies here.